production/kubernetes/build/templates/operator/main.jsonnet

```diff
@@ -1,7 +1,157 @@
 local k = import 'ksonnet-util/kausal.libsonnet';
-local agent_operator = import 'grafana-agent-operator/main.libsonnet';
+local secret = k.core.v1.secret;
+local pvc = k.core.v1.persistentVolumeClaim;
 
+local gen = import 'agent-operator-gen/main.libsonnet';
+local ga = gen.monitoring.v1alpha1.grafanaAgent;
+local mi = gen.monitoring.v1alpha1.metricsInstance;
+local li = gen.monitoring.v1alpha1.logsInstance;
+local pl = gen.monitoring.v1alpha1.podLogs;
+local int = gen.monitoring.v1alpha1.integration;
+
+local op = import 'grafana-agent-operator/operator.libsonnet';
+local ga_util = import 'grafana-agent-operator/util/grafana-agent.libsonnet';
+local mi_util =  import 'grafana-agent-operator/util/metricsinstance.libsonnet';
+local li_util = import 'grafana-agent-operator/util/logsinstance.libsonnet';
+local pl_util = import 'grafana-agent-operator/util/podlogs.libsonnet';
+local mon_util = import 'grafana-agent-operator/util/k8smonitors.libsonnet';
+local int_util = import 'grafana-agent-operator/util/integrations.libsonnet';
+
+local ksm = import 'kube-state-metrics/kube-state-metrics.libsonnet';
+
+## TODO: pod logs & metrics relabeling to match what we currently do / enable correlation
 {
-  agent_operator:
-    agent_operator.new(name='grafana-agent-operator', namespace='default')
+  local this = self,
+
+  _images:: {
+    agent: 'grafana/agent:v0.26.0-rc.0',
+    agent_operator: 'grafana/agent-operator:v0.26.0-rc.0',
+    ksm: 'registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.5.0'
+  },
+
+  _config:: {
+    namespace: '${NAMESPACE}',
+    metrics_url: '${METRICS_URL}',
+    metrics_user: '${METRICS_USER}',
+    metrics_key: '${METRICS_KEY}',
+    logs_url: '${LOGS_URL}',
+    logs_user: '${LOGS_USER}',
+    logs_key: '${LOGS_KEY}',
```

Comment:
I haven't seen this pattern before. Are we doing SED replace, or accepting environment variables?

Reply:
> Are we doing SED replace, or accepting environment variables?

the latter. generated manifests will have `${VAR_NAME}` style placeholders throughout that can be filled in using env vars and `envsubst` - example [here](https://github.com/grafana/agent/blob/main/production/kubernetes/agent-loki.yaml).

since this will be a more complex set of manifests, when i return i'll try to document this a bit better. the end user should not have to generate anything or use jsonnet, though.